routing/additional_edge.go

```diff
@@ -0,0 +1,113 @@
+// be appended to an existing route. Compared to normal edges of a route they
+// provide an explicit payload size function and are introduced because blinded
+// paths differ in their payload structure.
+type AdditionalEdge interface {
+	// PayloadSize returns the size of the payload for the additional edge.
+	PayloadSize(amount lnwire.MilliSatoshi, expiry uint32, legacy bool,
+		channelID uint64) uint64
+
+	// EdgePolicy returns the policy of the additional edge.
+	EdgePolicy() *models.CachedEdgePolicy
+}
+
+// PayloadSizeFunc defines the interface for the payload size function.
+type PayloadSizeFunc func(amount lnwire.MilliSatoshi, expiry uint32,
+	legacy bool, channelID uint64) uint64
+
+// PrivateEdge implements the AdditionalEdge interface. As the name implies it
+// is used for private route hints that the receiver adds for example to an
+// invoice.
+type PrivateEdge struct {
+	policy *models.CachedEdgePolicy
+}
+
+// EdgePolicy return the policy of the PrivateEdge.
+func (p *PrivateEdge) EdgePolicy() *models.CachedEdgePolicy {
+	return p.policy
+}
+
+// PayloadSize returns the sphinx payload size defined in BOLT04 if this edge
+// were to be included in a route.
+// NOTE: This payload calculation is only valid for intermediate hops of the
+// route.
+func (p *PrivateEdge) PayloadSize(amount lnwire.MilliSatoshi, expiry uint32,
+	legacy bool, channelID uint64) uint64 {
+
+	hop := route.Hop{
+		AmtToForward:     amount,
+		OutgoingTimeLock: expiry,
+		LegacyPayload:    legacy,
+	}
+
+	return hop.PayloadSize(channelID)
+}
+
+// BlindedEdge implements the AdditionalEdge interface. Blinded hops are viewed
+// as additional edges because they are appened at the end of a normal route.
+type BlindedEdge struct {
+	policy        *models.CachedEdgePolicy
+	cipherText    []byte
+	blindingPoint *btcec.PublicKey
+}
+
+// EdgePolicy return the policy of the BlindedEdge.
+func (b *BlindedEdge) EdgePolicy() *models.CachedEdgePolicy {
+	return b.policy
+}
+
+// PayloadSize returns the sphinx payload size defined in BOLT04 if this edge
+// were to be included in a route.
+// NOTE: This payload calculation is only valid for intermediate hops of the
+// blinded route.
+func (b *BlindedEdge) PayloadSize(_ lnwire.MilliSatoshi, _ uint32,
+	_ bool, _ uint64) uint64 {
+
+	hop := route.Hop{
+		BlindingPoint: b.blindingPoint,
+		LegacyPayload: false,
+		EncryptedData: b.cipherText,
+	}
+
+	// For blinded paths the next chanID is in the encrypted data tlv.
+	return hop.PayloadSize(0)
+}
+
+// Compile-time constraints to ensure the PrivateEdge and the BlindedEdge
+// implement the AdditionalEdge interface.
+var _ AdditionalEdge = (*PrivateEdge)(nil)
+var _ AdditionalEdge = (*BlindedEdge)(nil)
+
+// defaultHopPayloadSize is the default payload size of a normal (not-blinded)
+// hop in the route.
+// NOTE: This payload calculation is only valid for intermediate hops of the
+// route.
+func defaultHopPayloadSize(amount lnwire.MilliSatoshi, expiry uint32,
```

Comment:
Meant the reverse here that private edge impl can just call this (so that we don't have to create an empty edge: 
```
func (p *PrivateEdge) IntermediatePayloadSize(amount lnwire.MilliSatoshi,
	expiry uint32, legacy bool, channelID uint64) uint64 {

     return defaultHopPayloadSize(amount, expiry, legacy, channelID)
}

func defaultHopPayloadSize(amount lnwire.MilliSatoshi, expiry uint32,
	legacy bool, channelID uint64) uint64 {
	hop := route.Hop{
		AmtToForward:     amount,
		OutgoingTimeLock: expiry,
		LegacyPayload:    legacy,
	}

	return hop.PayloadSize(channelID)
}
```